static/app/components/idBadge/teamBadge/index.tsx

```diff
@@ -1,57 +1,40 @@
 import * as React from 'react';
-import isEqual from 'lodash/isEqual';
 
 import TeamStore from 'sentry/stores/teamStore';
 import {Team} from 'sentry/types';
 
-import Badge from './badge';
+import Badge, {BadgeProps} from './badge';
 
-type Props = React.ComponentProps<typeof Badge>;
+function TeamBadge(props: BadgeProps) {
+  const [team, setTeam] = React.useState<Team>(props.team);
 
-type State = {
-  team: Team;
-};
+  const onTeamStoreUpdate = React.useCallback(
+    (updatedTeam: Set<string>) => {
+      if (!updatedTeam.has(team.id)) {
+        return;
+      }
 
-class TeamBadgeContainer extends React.Component<Props, State> {
-  state: State = {team: this.props.team};
+      const newTeam = TeamStore.getById(team.id);
 
-  UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    if (this.state.team === nextProps.team) {
-      return;
-    }
+      if (!newTeam) {
+        return;
+      }
 
-    if (isEqual(this.state.team, nextProps.team)) {
-      return;
-    }
-
-    this.setState({team: nextProps.team});
-  }
-
-  componentWillUnmount() {
-    this.unlistener?.();
-  }
-
-  unlistener = TeamStore.listen(
-    (team: Set<string>) => this.onTeamStoreUpdate(team),
-    undefined
+      setTeam(newTeam);
+    },
+    [team, TeamStore]
   );
 
-  onTeamStoreUpdate(updatedTeam: Set<string>) {
-    if (!updatedTeam.has(this.state.team.id)) {
-      return;
-    }
-
-    const team = TeamStore.getById(this.state.team.id);
-    if (!team || isEqual(team.avatar, this.state.team.avatar)) {
```

Comment:
Does anyone know why we were comparing just the avatar obj here? Does my change to just always update it make sense? Imo unless we render a lot of these, and have some frequent changes, there should be no perf concerns on updating these everytime

```tsx
const onTeamStoreUpdate = React.useCallback(
    (updatedTeam: Set<string>) => {
      if (!updatedTeam.has(team.id)) {
        return;
      }

      const newTeam = TeamStore.getById(team.id);

      if (!newTeam) {
        return;
      }

      setTeam(newTeam);
    },
    [team, TeamStore]
  );
```

Reply:
Comparing only the avatar is likely to avoid re-renders when the visual elements aren't changing. I agree that it is reasonably safe to just always update.